include/openPMD/RecordComponent.hpp

```diff
@@ -434,6 +434,11 @@ class RecordComponent : public BaseRecordComponent
     std::shared_ptr<internal::RecordComponentData> m_recordComponentData{
         new internal::RecordComponentData()};
 
+    template <typename T>
+    void verifyChunk(Offset const &, Extent const &) const;
+
+    void verifyChunk(Datatype, Offset const &, Extent const &) const;
```

Comment:
Earlier versions of this PR added additional calls to `RecordComponent::storeChunk`, so instead of duplicating the verification logic, I extracted it. 
Now no longer necessary, but it's still a useful refactoring.